Documentation/platforms/arm/a64/boards/pinephone/index.rst

```diff
@@ -155,6 +155,13 @@ Supports all the features in ``lcd``,
 plus LVGL Graphics Library and Touch Panel (GT917S).
 Serial Console is enabled on UART0 at 115.2 kbps.
 
+modem
+-----
+
+Supports LTE Modem (EG25-G). Accepts AT Commands on UART3
+at ``/dev/ttyS1``.
```

Comment:
should we make /dev/ttySx match the datasheet number(UARTx)?

Reply:
Unfortunately UARTx won't match ttySx on PinePhone. Here are the upcoming UART ports:

- UART0 (ttyS0): Serial Console
- UART1 (ttyS1): Bluetooth Module (Realtek RTL8723CS, not integrated with NuttX yet)
- UART2: Unused, wired to Light Sensor and Compass Sensor
- UART3 (ttyS2): LTE Modem (Quectel EG25-G, integrated with NuttX)

[More details here](https://lupyuen.github.io/articles/serial#appendix-uart-ports-on-pinephone)

I could modify `modem/defconfig` to enable UART1 on ttyS1 (Bluetooth). But UART3 will always map to ttyS2 (LTE Modem) because UART2 is unused.

Shall I make this change? Thanks.

Reply:
It's better to adjust the tty driver register code to skip the unused hardware and dev/ttySx at the same time, or always register all possible tty devices even some mayn't use in the current state.

Reply:
Thanks I'll disable UART2 and /dev/ttyS2 for PinePhone, so that UART3 will map correctly to /dev/ttyS3. This should make our Modem Apps a lot easier to find the right /dev/tty.